src/common/include/com/amazonaws/kinesis/video/common/CommonDefs.h

```diff
@@ -1032,12 +1034,13 @@ extern PUBLIC_API atomicXor globalAtomicXor;
 //
 // Thread functionality
 //
-#define THREAD_CREATE      globalCreateThread
-#define THREAD_JOIN        globalJoinThread
-#define THREAD_SLEEP       globalThreadSleep
-#define THREAD_SLEEP_UNTIL globalThreadSleepUntil
-#define THREAD_CANCEL      globalCancelThread
-#define THREAD_DETACH      globalDetachThread
+#define THREAD_CREATE             globalCreateThread
+#define THREAD_CREATE_WITH_PARAMS globalCreateThreadWithParams

```

Comment:
There are some differing view points on how we should handle errors.  On one hand you can take the approach (which you've taken) that you're going to pass the stack size requested by the client straight to the lower level APIs (whatever platform specific ones there are) and just return/log an error if they fail. A downside to this approach is it makes testing the behavior, in case an unsupported value is provided for a particular platform, impossible without knowing what values may be valid/invalid for that particular platform. An alternative approach could test an array of values and in the tests we if/def per platform to have the correct assert (success vs failure) for that platform.  

If we are going to move forward with the understanding that we're not going to do any platform specific validation rather just provide the parameter to the underlying threading library and log any errors, then we should at least make that very clear in the documentation that this is the behavior of the API.  Specifically that we will attempt to use the facilities provided by the threading library for your platform to set the stack size if there are errors returned we will surface them via logs and/or on some platforms it may cause a crash.  If you wish to avoid this, is up to you (the client) to make sure you're specifying stack sizes which are supported on your platform.